src/lib/Transport/index.ts

```diff
@@ -80,66 +80,81 @@ export default class Transport {
     });
   }
 
-  
-  build(req: Request): Promise<request.CoreOptions> {
-    if (!this.isValidRequest(req.url)) {
-      const err = new Error('url provided is not a valid domo app endpoint');
-      
-      return Promise.reject(err);
-    }
-    
+  createContext(): Promise {
+    const options = {
+      method: 'POST',
+      url: `${this.client.server}/domoapps/apps/v2/contexts`,
+      json: { designId: this.manifest.id, mapping: this.manifest.mapping },
+      headers: this.client.getAuthHeader(),
+    };
+
+    return new Promise((resolve, reject) => {
+      request(options, (error, response, body) => {
+        if (error) reject(error);
+
+        if (response.statusCode !== 200) reject(response);
+
+        resolve(body[0]);
+      });
+    });
+  }
+
+  build(req: IncomingMessage): Promise<request.CoreOptions> {
     let api: string;
-    
+
     return this.domainPromise
       .then((domain) => {
         api = `${domain}${req.url}`;
-        
+
         return this.createContext();
       })
-      .then((context) => {
+      .then((context, body) => {
         const jar = request.jar();
-        
+
         const referer = (req.headers.referer.indexOf('?') >= 0)
           ? (`${req.headers.referer}&context=${context.id}`)
           : (`${req.headers.referer}?userId=27&customer=dev&locale=en-US&platform=desktop&context=${context.id}`);
-        
+
         const headers = {
-          ...req.headers,
           ...this.client.getAuthHeader(),
-          referer,
+          'Content-Type': req.headers['content-type'] || req.headers['Content-Type'] || 'application/json',
```

Comment:
Do we want to add a default for content-type?

Reply:
This is a carry over from the old server, but when I attempted to remove it the app service blew up so we added it back in for now.

Reply:
I think this part is fine `req.headers['content-type'] || req.headers['Content-Type']`, just wondering if we need `'application/json'`

Reply:
I think I found the offending header... `host`. I've removed that from the original req and pass everything else through the proxied request. It's working on `GET` requests. I don't have a way to test POST requests at the moment.